packages/grid/_modules_/grid/components/editCell/StringEditCell.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import TextField, { TextFieldProps } from '@material-ui/core/TextField';
+import { CellParams } from '../../models/params/cellParams';
+import { isDate } from '../../utils/utils';
+
+// export interface StringEditCellProps extends CellParams {
+//   isValid?: boolean;
+// }
+function mapColDefTypeToInputType(type: string) {
+  switch (type) {
+    case 'string':
+      return 'text';
+    case 'number':
+    case 'date':
+      return type;
+    case 'dateTime':
+      return 'datetime-local';
+    default:
+      return 'text';
+  }
+}
+export function StringEditCell(props: CellParams & TextFieldProps) {
+  const {
+    value,
+    api,
+    field,
+    row,
+    colDef,
+    getValue,
+    rowIndex,
+    colIndex,
+    isEditable,
+    ...textFieldProps
+  } = props;
+  const [inputValueState, setInputValueState] = React.useState(value || '');
+
+  const onValueChange = React.useCallback(
+    (event) => {
+      const newValue = event.target.value;
+      //TODO consider removing local state, and just use gridState
```

Comment:
What's the advantage? A local state seems a lot faster for performance

Reply:
allows to store it in the gridState, and intercept to apply validation for example, see demo.